eofconvention.fsx

```diff
@@ -0,0 +1,29 @@
+open System.IO
+
+let EOFIsEOL(path: FileInfo) = 
+    try
+        use sr = new StreamReader (path.FullName)
+        let filetext = sr.ReadToEnd()
+        (filetext |> Seq.last = '\n')
+        
+    with
+        | :? System.ArgumentException -> printfn "%A" path; true
+        
+
+let EOFIsNotEOL(path: FileInfo) = not (EOFIsEOL path)
+
+let NotPrefix (value: string) (str: string) =
+    not (str.StartsWith(value))
+
+
+let invalidFiles = 
+    Directory.GetFiles(".", "*.*", SearchOption.AllDirectories) 
+    |> Array.filter (NotPrefix "./node_modules")
+    |> Array.filter (NotPrefix "./.git")
+    |> Array.map (fun x -> FileInfo(x))
+    |> Array.filter EOFIsNotEOL
```

Comment:
@realmarv I think you can use Seq. functions here instead of Array. ones; this way, if in the future someone changes line 20 to return a list instead of an array, then you wouldn't need the cascading changes (because Seq. functions work both on lists and also arrays)

Reply:
Done